main.js

```diff
@@ -1,51 +1,126 @@
+const nowPlayingUrl = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apiKey}`
 const popularUrl = `https://api.themoviedb.org/3/movie/popular?api_key=${apiKey}`
 const topRatedUrl = `https://api.themoviedb.org/3/movie/top_rated?api_key=${apiKey}`
 const upcomingUrl = `https://api.themoviedb.org/3/movie/upcoming?api_key=${apiKey}`
 
-
 // search movies
 const searchInput = document.getElementById("inputValue")
-const submitButton = document.getElementById('submitButton')
-
-submitButton.addEventListener('click', function() {
-    let searchedList = document.getElementById("searchedList")
-    searchedList.innerHTML = ''
-    const movieName = searchInput.value;
-    const searchUrl = searchMovieUrl + movieName
-
-    createPlaylist(searchUrl, 'searched')
+const submitButton = document.getElementById("submitButton")
+if (submitButton) {
+   submitButton.addEventListener("click", function () {
+      let searchedList = document.getElementById("searchedList")
+      searchedList.innerHTML = ""
+      const movieName = searchInput.value
+      const searchUrl = searchMovieUrl + movieName
+      createPlaylist(searchUrl, "searched")
+      searchInput.value = ""
+   })
+}
+/////////////////////////////////////////////BGINNING OF PLAYLIST CREATOR/////////////////////////////////////////////
+// getting elements and creating global variables
+let listNameTxt = document.getElementById("listNameTxt")
+let submitListInfoBtn = document.getElementById("submitListInfoBtn")
+let listBuilderInfoDiv = document.getElementById("listBuilderInfoDiv")
+let addFilmsDiv = document.getElementById("addFilmsDiv")
+let addFilmsSearchBtn = document.getElementById("addFilmsSearchBtn")
+let addFilmsSearchTxt = document.getElementById("addFilmsSearchTxt")
+let searchResultsDiv = document.getElementById("searchResultsDiv")
+let addFilmsToListBtn = document.getElementById("addFilmsToListBtn")
+var listName
+var tempArr = []
+
+// first window - gets list name from user
+submitListInfoBtn.addEventListener("click", function () {
+   listBuilderInfoDiv.style.display = "none"
+   addFilmsDiv.style.display = "flex"
+   listName = listNameTxt.value
+   // creates an empty collection and doc to hold the list
+   firebase.auth().onAuthStateChanged(function (user) {
+      if (user) {
+         let userId = user.uid
+         db.collection("users")
+            .doc(user.uid)
+            .collection("playlists")
+            .doc(listName)
+            .set({})
+      }
+   })
+})
 
-    
-    searchInput.value = ''
+// search button
+addFilmsSearchBtn.addEventListener("click", function () {
+   searchResultsDiv.innerHTML = ""
+   let userInput = addFilmsSearchTxt.value
+   let searchUrl = searchMovieUrl + userInput
+   // print list of movies
+   fetch(searchUrl)
+      .then((res) => res.json())
+      .then((data) => {
+         let movieArray = data.results
+         console.log(movieArray)
+         for (let i = 0; i < movieArray.length; i++) {
```

Comment:
so this part is creating a movie poster for each search option correct?

Reply:
i feel like for the initial search generating the movieArray should be returned as an ordered array of objects that we can access and filter through for when the userid matches